app/src/App/Domain/Service/UsersService.php

```diff
@@ -36,7 +36,13 @@ public function getAll(): array
 
     public function getById(UserId $id): ?User
     {
-        return $this->users->findById($id);
+        $user = $this->users->findById($id);
+
+        if (!$user instanceof User) {
+            throw UserNotFoundException::fromUsername($username);
```

Comment:
Yep, I was just about to make a new commit with it ...been hit too many times in the past to ignore https://en.wikipedia.org/wiki/Confirmation_bias :-)